jolt-core/src/jolt/vm/instruction_lookups.rs

```diff
@@ -122,7 +122,7 @@ where
     fn batch(&self) -> Self::BatchedPolynomials {
         use rayon::prelude::*;
         let (batched_dim_read, (batched_final, batched_E, batched_flag)) = rayon::join(
-            || DensePolynomial::merge_dual(self.dim.as_ref(), self.read_cts.as_ref()),
+            || DensePolynomial::merge(&[self.dim.as_slice(), self.read_cts.as_slice()].concat()),
```

Comment:
Can we drop the concat here?

I'm concerned that we're going to allocate a vector of length `self.dim.len() + self.read_cts.len()` twice. Once after the `.concat()` call and once in `DensePolynomial::merge`. Recall that these are likely to be massive (2^28 -> 2^32).